controllers/iamauthpolicy_controller.go

```diff
@@ -47,17 +60,9 @@ func (c *IAMAuthPolicyController) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 	c.log.Infow("reconcile", "req", req, "targetRef", k8sPolicy.Spec.TargetRef)
 
```

Comment:
Need some guard clauses that similar to these?

https://github.com/aws/aws-application-networking-k8s/blob/a2d5b973b3516f11784d1a764830e8de2eadf4a7/controllers/accesslogpolicy_controller.go#L129-L145

Reply:
I can add group check, kind check is already there. https://github.com/aws/aws-application-networking-k8s/blob/main/controllers/iamauthpolicy_controller.go#L61-L77